commit 4369f441ce52d131957e9992a356c63aa66e2df4
Author: Richard Hughes <richard@hughsie.com>
Date:   Tue Apr 6 16:24:54 2010 +0100

    Ensure we process font requests by correctly processing the session ignore blacklist

diff --git a/src/gpk-dbus-task.c b/src/gpk-dbus-task.c
index 3ba46c5..6dc9101 100644
--- a/src/gpk-dbus-task.c
+++ b/src/gpk-dbus-task.c
@@ -2141,25 +2141,31 @@ out:
 
 /**
  * gpk_dbus_task_install_check_exec_ignored:
+ *
+ * Returns %FALSE if the executed program is in the ignored list.
  **/
 static gboolean
 gpk_dbus_task_install_check_exec_ignored (GpkDbusTask *dtask)
 {
 	gchar *ignored_str;
 	gchar **ignored = NULL;
-	gboolean ret = FALSE;
+	gboolean ret = TRUE;
+	GError *error = NULL;
 	guint i;
 
 	/* check it's not session wide banned in gconf */
-	ignored_str = gconf_client_get_string (dtask->priv->gconf_client, GPK_CONF_IGNORED_DBUS_REQUESTS, NULL);
-	if (ignored_str == NULL)
+	ignored_str = gconf_client_get_string (dtask->priv->gconf_client, GPK_CONF_IGNORED_DBUS_REQUESTS, &error);
+	if (ignored_str == NULL) {
+		egg_warning ("failed to get ignored requests: %s", error->message);
+		g_error_free (error);
 		goto out;
+	}
 
 	/* check each one */
 	ignored = g_strsplit (ignored_str, ",", -1);
 	for (i=0; ignored[i] != NULL; i++) {
 		if (g_strcmp0 (dtask->priv->exec, ignored[i]) == 0) {
-			ret = TRUE;
+			ret = FALSE;
 			break;
 		}
 	}
commit 2513476d3242952a12610bcc898ae1cc13f4984e
Author: Richard Hughes <richard@hughsie.com>
Date:   Tue Apr 6 16:37:54 2010 +0100

    Use /proc/%i/exe instead of cmdline when getting the exec of a sender

diff --git a/src/gpk-dbus.c b/src/gpk-dbus.c
index e7f603f..f775a22 100644
--- a/src/gpk-dbus.c
+++ b/src/gpk-dbus.c
@@ -207,7 +207,6 @@ out:
 static gchar *
 gpk_dbus_get_exec_for_sender (GpkDbus *dbus, const gchar *sender)
 {
-	gboolean ret;
 	gchar *filename = NULL;
 	gchar *cmdline = NULL;
 	GError *error = NULL;
@@ -224,19 +223,12 @@ gpk_dbus_get_exec_for_sender (GpkDbus *dbus, const gchar *sender)
 	}
 
 	/* get command line from proc */
-	filename = g_strdup_printf ("/proc/%i/cmdline", pid);
-	ret = g_file_get_contents (filename, &cmdline, NULL, &error);
-	if (!ret) {
-		egg_warning ("failed to get cmdline: %s", error->message);
+	filename = g_strdup_printf ("/proc/%i/exe", pid);
+	cmdline = g_file_read_link (filename, &error);
+	if (cmdline == NULL) {
+		egg_warning ("failed to find exec: %s", error->message);
 		g_error_free (error);
 	}
-
-	/* if command line contains (deleted) the original binary is invalid */
-	if (g_strstr_len (cmdline, -1, "(deleted)") != NULL) {
-		g_free (cmdline);
-		cmdline = NULL;
-		goto out;
-	}
 out:
 	g_free (filename);
 	return cmdline;
commit 3e1fada25f17abf73a548ec1462f02abdb439d46
Author: Richard Hughes <richard@hughsie.com>
Date:   Tue Apr 6 16:38:13 2010 +0100

    Ensure gpk_dbus_task_get_package_for_exec() returns a package name rather than a package_id

diff --git a/src/gpk-dbus-task.c b/src/gpk-dbus-task.c
index 6dc9101..45440b7 100644
--- a/src/gpk-dbus-task.c
+++ b/src/gpk-dbus-task.c
@@ -2898,12 +2898,14 @@ out:
 static gchar *
 gpk_dbus_task_get_package_for_exec (GpkDbusTask *dtask, const gchar *exec)
 {
+	const gchar *package_id;
 	gchar *package = NULL;
 	GError *error = NULL;
 	GPtrArray *array = NULL;
 	PkPackage *item;
 	PkResults *results = NULL;
 	gchar **values = NULL;
+	gchar **split = NULL;
 
 	/* find the package name */
 	values = g_strsplit (exec, "&", -1);
@@ -2930,12 +2932,13 @@ gpk_dbus_task_get_package_for_exec (GpkDbusTask *dtask, const gchar *exec)
 
 	/* copy name */
 	item = g_ptr_array_index (array, 0);
-	g_object_get (item,
-		      "package-id", &package,
-		      NULL);
+	package_id = pk_package_get_id (item);
+	split = pk_package_id_split (package_id);
+	package = g_strdup (split[0]);
 	egg_debug ("got package %s", package);
 out:
 	g_strfreev (values);
+	g_strfreev (split);
 	if (array != NULL)
 		g_ptr_array_unref (array);
 	if (results != NULL)
